app/src/main/java/com/teamltt/carcare/database/DbHelper.java

```diff
@@ -238,6 +314,81 @@ public int updateVehicle(long vehicleId, Vehicle vehicle) {
+        if (status != DbHelper.DB_OK) {
+            return status;
+        }
+        if (reminderId < 0) {
+            return -1;
+        }
+        int numAffected = ReminderContract.update(db, reminderId, vehicleId, name, featureId, comparison, value, date);
+        db.close();
+        return numAffected;
+    }
+
+    //takes a vehicle id, returns a list of all reminders associated with that vehicle
+    public List<Reminder> getRemindersByVehicleId(long vehicleId) {
+        SQLiteDatabase db = getReadableDatabase();
+        Cursor cursor = ReminderContract.queryByVehicleId(db, vehicleId);
+        List<Reminder> reminders = new ArrayList<>();
+        while (cursor.moveToNext()) {
+            long reminderId = cursor.getLong(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_ID));
+            String name = getCursorColumn(cursor, ReminderContract.ReminderEntry.COLUMN_NAME_NAME);
+            int featureId = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_FEATURE_ID));
+            int comparison = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_COMPARISON));
+            int value = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_VALUE));
+            String date = getCursorColumn(cursor, ReminderContract.ReminderEntry.COLUMN_NAME_DATE);
+            reminders.add(new Reminder(reminderId, name, featureId, comparison, value, date));
+        }
+        cursor.close();
+        db.close();
+        return reminders;
+    }
+
+    //takes a reminder id (the integer identifier from the table) and returns a Reminder object with the
+    //values associated with that reminder
+    public Reminder getReminderByReminderId(long reminderId) {
+        SQLiteDatabase db = getReadableDatabase();
+        Cursor cursor = ReminderContract.queryByReminderId(db, reminderId);
+        cursor.moveToFirst();
+        String name = cursor.getString(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_NAME));
+        int featureId = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_FEATURE_ID));
+        int comparison = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_COMPARISON));
+        int value = cursor.getInt(cursor.getColumnIndexOrThrow(ReminderContract.ReminderEntry.COLUMN_NAME_VALUE));
+        String date = getCursorColumn(cursor, ReminderContract.ReminderEntry.COLUMN_NAME_DATE);
+        Reminder reminder = new Reminder(reminderId, name, featureId, comparison, value, date);
+        return reminder;
+
+    }
+
+    public long deleteReminder(long reminderId) {
+        SQLiteDatabase db = getWritableDatabase();
+        long status = DbHelper.errorChecks(db);
+        if (status != DbHelper.DB_OK) {
+            return status;
+        }
+        if (reminderId < 0) {
+            return -1;
```

Comment:
Nothing, if you pass a reminder of id -1 it won't exist in the database so I just return -1, but I could do anything else there.